### ISSUE TITLE
Cannot choose audio or subtitle track when transcoding video

### ISSUE THREAD

**Issue by truenicoco** [jellyfin/jellyfin-kodi#360]
**Describe the bug**
Without transcoding, I can properly choose between audio and subtitles tracks from the kodi interface. However, when activating transcoding, only one (the first I guess) audio track is available. And there is no subtitle available.

**To Reproduce**
1. In settings, activate transcoding for HEVC (needed for my raspberry pi 3)
2. Launch a HEVC movie
3. Go to audio and subtitles options
4. See that there is no choice

**Expected behavior**
It should offer the different audio tracks and subtitles, as the web client does, even when transcoding is on.

**Logs**
[Kodi logs](https://paste.debian.net/hidden/6c0e24fa/)
[Jellyfin-ffmpeg logs](https://paste.debian.net/hidden/dbf1e205/)

**System (please complete the following information):**
 - Jellyfin Version: 10.6.2 on debian buster x86, from the jellyfin repo
 - Kodi Version: 18.7 on raspbian, raspberry pi 3
 - Addon Version: 0.5.8
 - Playback Mode: Add-On

**Comment by Ashyni** [comment]
Hello, in order to choose the audio/sub track, you need to Enable audio/subtitles selection in Jellyfin Settings under Playback tab.

**Comment by truenicoco** [comment]
@Ashyni I don't find such options in Jellyfin settings. Are you sure this is what cause my issue? Audio/sub selection works:

- in kodi without transcoding
- in jellyfin web client with or without transcoding

So this makes me think the problem is on the kodi add-on side… but I might be wrong.

**Comment by Ashyni** [comment]
Yes i was talking about the jellyfin-kodi add-on settings.
![selection-settings](https://user-images.githubusercontent.com/18462848/90153552-6e5bf600-dd89-11ea-8a45-06fcf37b3b17.PNG)

**Comment by truenicoco** [comment]
You're right! Sorry for doubting you. It is just counter intuitive than even with this option set to disabled, one can still choose audio/subs when no transcoding occurs.

**Comment by mcarlton00** [comment]
It's a bit weird, but we're kinda working around the restrictions of Kodi in this case.

When we're transcoding here, the subs get burned into the video stream so all Kodi sees is a single video stream with no supplemental audio or subtitle streams it can poll.  That's why those options tend to be greyed out during transcoding.  However during direct playing we can both read the raw file for embedded streams and request additional external streams that the server may have.  In order to change the subtitles during playback in this case, it would need to stop playback entirely, stopping the ffmpeg transcoding session on the server, and then start a new one with the new media streams enabled.

Presumably we should be able to transcode the video stream but still have the option of choosing an external sub file during playback, but I haven't looked into how feasible it is

**Comment by bzzbrr** [comment]
I don't know how difficult this would be, but if you're modifying the selection it would be great to have the options for Enable audio/subtitles selection be:

- Enabled
- Disabled
- Subtitles Only
- Audio Only
- **Only When Transcoding**

I personally don't want the pop-up when direct playing because it can always be changed while playing, but for transcoding it has to be done before.

**Comment by Electry** [comment]
@bzzbrr The popup won't show if you're direct playing. The option 'Enable audio/subtitles' is only relevant to transcoding, where the server either burns the embedded subs, or sends an external text stream to kodi.

When direct playing, you are only able to select audio/subs stream in the kodi video player OSD.

**Comment by theAeon** [comment]
Possibly related, but when I select a burned-in subtitle on a file that will be transcoded it fails to play entirely. The reason I say it might be related is that the internal subtitles do not show as in the OP if you do not select them.
